File: Shajee_CS280 HW#4/Complex.java
import java.util.*;
public class Complex{
   double real;
   double imag;
   public Complex() {

        this(0.0,0.0);
    }
    public Complex(double real) {

        this(real, 0.0);
    }
    public Complex(double real, double imag) {

        this.real = real;
        this.imag = imag;
    }
    public Complex add(Complex o) {

        return new Complex(real + o.real , imag + o.imag);
    }
	
	public Complex add(double o){
		return new Complex(real+o,imag);
	}
   public String div(Complex o) {
			double num =(real*o.real)+((-1)*(imag*((-1)*(o.imag))));
			double num1 =(real*((-1)*(o.imag))+(imag*o.real));
			
			double den =(o.real*o.real);
			double den1 =(o.imag*((-1)*(o.imag)))*(-1);
				
			den=den+den1;
				
        return new Complex((num/den),(num1/den)).toString();
        }
   public String div(double o){
        return new Complex(real/o,imag/o).toString();
	}

    public Complex mul(Complex o) {
        return new Complex((real*o.real)+((-1)*(imag*o.imag)),(real*o.imag)+(imag*o.real) );
    }
	
	public Complex mul(double o){
		return new Complex((real*o),(imag*o)); 
	}

    public Complex sub(Complex o) {

        return new Complex(real - o.real , imag - o.imag);
    }
	public Complex sub(double o){
		return new Complex(real-o,imag);
	}
    public boolean isDoubleInt(double d)
    {
      return Math.abs(Math.floor(d) - d) == (double)0;
    }

    public String toString() {
       if(isDoubleInt(this.real)==true&&isDoubleInt(this.imag)==true){
         if (this.imag < 0) {
	          
			   return "("+(int)real + "" + (int)imag + "j)";
	     
		   } else { 
		    
			   return "("+(int)real + "+" + (int)imag + "j)";
		   }
       }
       else{
         if(isDoubleInt(this.real)==true&&isDoubleInt(this.imag)==false){
            if (this.imag < 0) {
	          
			         return "("+(int)real + "" + imag + "j)";
	     
		         } else { 
		    
			         return "("+(int)real + "+" + imag + "j)";
		         }
         }
         else{
            if(isDoubleInt(this.real)==false&&isDoubleInt(this.imag)==true){
               if (this.imag < 0) {
	          
			         return "("+real + "" + (int)imag + "j)";
	     
		         } else { 
		    
			         return "("+real + "+" + (int)imag + "j)";
		         }
            }
            else{
               if (this.imag < 0) {
	          
			         return "("+real + "" + imag + "j)";
	     
		         } else { 
		    
			         return "("+real + "+" + imag + "j)";
		         }
            }
         }
       }
	 }
}

File: Shajee_CS280 HW#4/complex.cc
#include <iostream>
#include "complex.h"

complex::complex(double real, double imag) : real(real), imag(imag) {}

complex complex::operator+(const complex &o) const {

    return complex(real + o.real,imag+ o.imag);
}

complex complex::operator+(double n) const {

    return complex(real + n , imag);
}

complex complex::operator-(const complex &o) const {

    return complex(real - o.real , imag - o.imag);
}

complex complex::operator-(double n) const {

    return complex(real - n , imag);
}

complex complex::operator*(const complex &o) const {

    return complex((real * o.real)+((-1)*imag * o.imag),((real*o.imag)+(imag*o.real)));
}

complex complex::operator*(double n) const {

    return complex(real * n, n*imag);
}

complex complex::operator/(const complex &o) const {
  double  den =(o.real * o.real)+((-1)*(o.imag*(o.imag*(-1))));
    return complex((((real * o.real)+((-1)*imag * o.imag*(-1)))/(den)),((real*o.imag*(-1))+(imag*o.real))/den);
}

complex complex::operator/(double n) const {

    return complex(real/n, imag /n);
}


complex operator+(double n, const complex &o) {

    return o + n;
}

complex operator-(double n, const complex &o) {

    return complex(n) - o;
}

complex operator*(double n, const complex &o) {

    return o * n;
}

complex operator/(double n, const complex &o) {

    return complex(n) / o;
}

ostream &operator<<(ostream &out, const complex &o) {

    if(o.imag>=0&&o.real>=0){
      out <<"("<<(double) o.real << "+" << (double)o.imag << "j)";
      return out;
    }
    else{
      if(o.imag>=0&&o.real<=0){
         out<<"(" <<(double) o.real << "+" << (double)o.imag << "j)";
         return out;
      }
      else{
         out <<"("<< (double)o.real << (double)o.imag << "j)";
         return out;
      }
    }
}